Sources/SwiftParser/Expressions.swift

```diff
@@ -1199,6 +1207,10 @@ extension Parser {
     pattern: PatternContext,
     flavor: ExprFlavor
   ) -> RawExprSyntax {
+    if self.isAtModuleSelector() && (!pattern.admitsBinding || !self.lookahead().isInBindingPatternPosition()) {
+      return self.parseIdentifierExpression(flavor: flavor)
+    }
```

Comment:
Could you add comments why this is hoisted?

Reply:
Its primary purpose is to ensure that invalid module selectors are handled correctly (although valid ones went through there too), but now that we allow a much narrower set of invalid tokens, we can just handle them all in a `switch` case, which will be clearer. I'll do that instead.